services/director/Dockerfile

```diff
@@ -1,106 +1,143 @@
+ENV PATH "/home/scu/.local/bin:$PATH"
 
-ENV HOME /home/scu
-# TODO: tmp workaround for https://github.com/pypa/pip/issues/6197
-# - Added explicit --cache-dir that is removed at the end of the build
-ENV PIP  /home/scu/.venv/bin/pip3 --cache-dir=/tmp/.pipcache
-# -----
+# All MY_ variables are customized
+ENV MY_PIP pip3 --no-cache-dir
+ENV MY_BUILD_TARGET base
 
-EXPOSE 8001
+# TODO: update environs
 ENV REGISTRY_AUTH = ''
 ENV REGISTRY_USER = ''
 ENV REGISTRY_PW = ''
 ENV REGISTRY_URL = ''
 ENV PUBLISHED_HOST_NAME=''
 ENV EXTRA_HOSTS_SUFFIX = 'undefined'
 
-# ----------------------------------------------------------------
-COPY --chown=scu:scu scripts/docker/healthcheck_curl_host.py /healthcheck/healthcheck_curl_host.py
-# will start to run after interval every interval. fails after timeout. fail do not count if during start-period. will do # retries
-HEALTHCHECK --interval=10s --timeout=30s --start-period=20s --retries=3 CMD [ "python3", "/healthcheck/healthcheck_curl_host.py", "http://localhost:8001/v0/" ]
 
-WORKDIR /home/scu
+EXPOSE 8080
 
 # -------------------------- Build stage -------------------
+# Installs build/package management tools and third party dependencies
 #
-# - Preserves relative folder structure
+# + /build             WORKDIR
 #
-# + /home/scu/              $HOME
-#    + services/director
-#        ...
-#    + packages
-#        ...
+
 FROM base as build
 
+ENV MY_BUILD_TARGET build
+
 RUN apk add --no-cache \
     gcc \
     libc-dev \
     git
 
-# create virtual environment
-RUN python3 -m venv $HOME/.venv &&\
-    $PIP install --upgrade \
+RUN $MY_PIP install --upgrade \
       pip \
       wheel \
       setuptools
 
-# this is done to accelerate installation of director later on
-COPY --chown=scu:scu services/director/requirements/base.txt requirements-base.txt
-RUN $PIP install -r requirements-base.txt &&\
-    rm requirements-base.txt
+WORKDIR /build
+
+# install base 3rd party dependencies
+COPY  services/director/requirements/*.txt \
+      services/director/requirements/
+
+RUN $MY_PIP install \
+      -r services/director/requirements/base.txt
 
-COPY --chown=scu:scu services/director/docker docker
 
 # --------------------------Development stage -------------------
+# Source code accessible in host but runs in container
+# Runs as myu with same gid/uid as host
+# Placed at the end to speed-up the build if images targeting production
+#
+#  + /devel         WORKDIR
+#    + services  (mounted volume)
+#
 FROM build as development
 
-ARG HOST_GID_ARG=1000
+ENV MY_BUILD_TARGET development
+ENV MY_BOOT_MODE development
+
 
-# in dev-mode we give access to `scu` to host's mapped volumes
-# FIXME: files created by scu cannot be deleted by host! we need to do the same group in host?
-RUN addgroup -g $HOST_GID_ARG hgrp &&\
-    addgroup scu hgrp && \
-    chown -R scu:scu $HOME/.venv &&\
-    rm -rf /tmp/.pipcache
+#  TODO: consider installing tooling in a separate virtualenv?
```

Comment:
what is the meaning of this?

Reply:
The TODO actually do not belong here ... The idea is that sometimes we need tools (e.g. pip-compile or cookiecutter or linter) that "polute" with dependencies the py-environment where my app actually runs. Then it is desirable to isolate the env for the app and that of tooling